### ISSUE TITLE
service to service calls - grpc exception: upstream request timeout

### ISSUE THREAD

**Issue by matthiasng** [microsoft/azure-container-apps#956]
> Please provide us with the following information:
> ---------------------------------------------------------------

### This issue is a: (mark with an x)
- [x] bug report -> please search issues before submitting
- [ ] documentation issue or request
- [X] regression (a behavior that used to work and stopped in a new release)

### Issue description

Since yesterday we encounter some issues with service to service invocations (GRPC).

Service A calls a function on service B. The function on service B takes ~30 seconds to complete.

Service A logs:

```
2023-10-24T14:16:29.042157939Z [14:16:29 INF] Start processing HTTP request POST http://localhost:50001/serviceB.v1.Resource/Create
2023-10-24T14:16:29.042176939Z [14:16:29 INF] Sending HTTP request POST http://localhost:50001/serviceB.v1.Resource/Create
2023-10-24T14:16:44.043715358Z [14:16:44 INF] Received HTTP response headers after 15001.8853ms - 200
2023-10-24T14:16:44.043767059Z [14:16:44 INF] End processing HTTP request after 15002.2121ms - 200
2023-10-24T14:16:44.045029678Z [14:16:44 INF] Call failed with gRPC error status. Status code: 'Unavailable', Message: 'upstream request timeout'.
```

Service B logs:
```
2023-10-24T14:16:59.067959179Z [14:16:59 INF] Executed endpoint 'gRPC - /service.v1.Resource/Create'
2023-10-24T14:16:59.067965479Z [14:16:59 INF] HTTP POST /service.v1.Resource/Create responded 200 in 30016.9367 ms
2023-10-24T14:16:59.067972279Z [14:16:59 INF] Request finished HTTP/2 POST http://127.0.0.1:5001/serviceB.v1.Resource/Create application/grpc+proto - - 200 - application/grpc 30017.1584ms
```

As we can see from the logs, the function gets called on service B. Service B finishes succesfully but the caller side gets an Unavailable GRPC exception after 15 seconds.
My current assumption is that the sidecar [resliliency configuration](https://docs.dapr.io/operations/resiliency/policies/) has changed, sadly there is no way to see or change it.

I also cannot reproduce the issue locally (setup according to: https://docs.dapr.io/developing-applications/local-development/ides/vscode/vscode-how-to-debug-multiple-dapr-apps/) and we didn't changed anything on the container apps in the last days.

We have three different container apps environment (two on europe wets, one on us central) all the behaviour is the same.

### Steps to reproduce
1. Two services, Service A calls a function on service B
    - Add a sleep(30) to the handler in service B

**Expected behavior** The caller doens't get an "Unavailable" exception after 15 seconds

**Actual behavior** service A gets a succesfull response from service B

**Screenshots**         
-

### Additional context
- services are written in .Net
  - Dapr.AspNetCore Version="1.9.0"
  - Grpc.AspNetCore" Version="2.49.0"
- Requests that take less than 15 seconds work fine.

**Comment by ItalyPaleAle** [comment]
Could you please provide the logs from the "daprd" container around the time these failures happen?

**Comment by matthiasng** [comment]
I set the dapr log level to "debug" and also enabled API logging before reproducing the issue to get dapr logs.

**Service A dapr log**

```
2023-10-26T09:58:58.892461256Z time="2023-10-26T09:58:58.892321956Z" level=info msg="gRPC API Called" app_id=servicea instance=ABC method=/serviceB.v1.Resource/Create scope=dapr.runtime.grpc.api-info type=log useragent="grpc-dotnet/2.55.0 (.NET 6.0.24; CLR 6.0.24; net6.0; linux; x64)" ver=1.11.2-msft-3
```

**Service B dapr logs**

```
2023-10-26T09:59:19.500904004Z time="2023-10-26T09:59:19.5006617Z" level=debug msg="No active messages require lock renewal for queue some-queue" app_id=serviceB component="some-queue (bindings.azure.servicebusqueues/v1)" instance=abc scope=dapr.contrib type=log ver=1.11.2-msft-3
2023-10-26T09:59:19.500910204Z time="2023-10-26T09:59:19.5006726Z" level=debug msg="Renewing message locks for queue some-queue" app_id=serviceb component="some-queue (bindings.azure.servicebusqueues/v1)" instance=abc scope=dapr.contrib type=log ver=1.11.2-msft-3
```

On the side of serviceB there are only some log messages related to servicebus queues.

**Comment by take-4c** [comment]
Any updates? I also have met the same issue. 
When any container apps send a request to this container apps with gRPC via dapr, it behaves like a timeout after about 15 seconds, resulting in an error. I'm hoping for prompt resolution.

**Comment by matthiasng** [comment]
I created a support issue in the Azure Portal and they already fixed it 2 days ago. Sadly I don't have any details what exactly caused the problem (so it could be only a temporary fix for our specific container apps env) but it was definitely something on the Azure side.

For reference I also created a small example to reproduce the issue: https://github.com/matthiasng/azurecapp-test

One way to verify if you run in the same problems is to check if you consistently get a grpc error with status code "Unavailable" and message "upstream request timeout" after 15 sec. In case you get something like a "Deadline exceeded" it could be that you have set a timeout somewhere. At least that is what I observed during my tests.
